.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Run checks on pull requests and pushes on main branch
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
+
+jobs:
+  tests:
+    name: Checks of unit tests and format checks
+    runs-on: "${{ matrix.os }}"
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.7, 3.8, 3.9]
+        experimental: [false]
+        include:
+          - os: ubuntu-latest
+            path: ~/.cache/pypoetry
+          - os: windows-latest
+            path: ~\AppData\Local\pypoetry\Cache
+          - os: macos-latest
+            path: ~/Library/Caches/pypoetry
+    env:
+      PYTHON: ${{ matrix.python-version }}
+      OS: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Get full Python version
+        id: full-python-version
+        shell: bash
+        run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
+
+      - name: Bootstrap poetry
+        shell: bash
+        run: |
+          curl -sL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py \
+            | python - -y ${{ matrix.bootstrap-args }}
+      - name: Update PATH
+        if: ${{ matrix.os != 'windows-latest' }}
+        shell: bash
+        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: Update Path for Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        shell: bash
+        run: echo "$APPDATA\Python\Scripts" >> $GITHUB_PATH
+
+      - name: Configure poetry
+        shell: bash
+        run: poetry config virtualenvs.in-project true
+
+      - name: Set up cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: .venv
```

Comment:
You should make sure to also cache `${{ matrix.path }}`, since you've gone to the work of configuring it up above.